src/lib/autodiscovery/autodiscovery_manager.py

```diff
@@ -1,76 +1,148 @@
+        resource_mapping = {}
+        for extension in mapping_yaml["gcp_extension_mapping"]:
+            for service in extension["services"]:
+                for monitored_resource in service.get("monitored_resources", []):
+                    resource_name = monitored_resource.get("value")
+                    if resource_name is not None:
+                        resource_mapping.setdefault(resource_name, []).append(
+                            ServiceStub(
+                                extension_name=extension["extension_name"],
+                                service_name=service["service_name"],
+                                feature_set_name=service["feature_set"],
+                            )
+                        )
+        return {
+            resource_name: list(set(service_stub))
+            for resource_name, service_stub in resource_mapping.items()
+        }
+
+    async def check_resources_to_autodiscovery(
+        self, services: List[GCPService]
+    ) -> Dict[str, AutodiscoveryResourceLinking]:
+        prepared_resources = {}
+
+        enabled_services_identifiers = {}
+        disabled_services_identifiers = {}
+
+        for service in services:
+            # com.dynatrace.extension.<%=extensionName%>
+            extension_name = service.extension_name.split(".")[-1]
+            identifier = ServiceStub(extension_name, service.name, service.feature_set)
+            if service.is_enabled:
+                enabled_services_identifiers[identifier] = service
+            else:
+                disabled_services_identifiers[identifier] = service
+
+        for resource in self.autodiscovery_config["searched_resources"]:
+            if resource in self.autodiscovery_resource_mapping:
+                possible_linking = []
+                disabled_linking = []
+
+                for required_service in self.autodiscovery_resource_mapping[resource]:
+                    if required_service in enabled_services_identifiers:
+                        possible_linking.append(enabled_services_identifiers[required_service])
+                    if required_service in disabled_services_identifiers:
+                        disabled_linking.append(disabled_services_identifiers[required_service])
+
+                if possible_linking:
+                    prepared_resources[resource] = AutodiscoveryResourceLinking(
+                        possible_service_linking=possible_linking,
+                        disabled_services_for_resource=disabled_linking,
+                    )
+
+                else:
+                    self.logging_context.log(
+                        f"Can't add resource {resource}. Make sure if extension {self.autodiscovery_resource_mapping[resource][0].extension_name} with proper services is enabled."
+                    )
+
+            else:
+                prepared_resources[resource] = None
```

Comment:
What does this mean? If the resource is not in the mapping, you won't be looking for metrics from it? What about resources which we are not considering in any extensions?
If this is on purpose and it's to be extended, no problem. Next PR :)